<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::WindowModality::WindowModal</enum>
  </property>
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>320</width>
    <height>401</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">Calculator</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LayoutDirection::LeftToRight</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="font">
    <font>
     <stylestrategy>NoAntialias</stylestrategy>
     <kerning>false</kerning>
    </font>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LayoutDirection::LeftToRight</enum>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <property name="leftMargin">
     <number>4</number>
    </property>
    <property name="topMargin">
     <number>4</number>
    </property>
    <property name="rightMargin">
     <number>4</number>
    </property>
    <property name="bottomMargin">
     <number>4</number>
    </property>
    <property name="spacing">
     <number>4</number>
    </property>
    <item row="2" column="0">
     <widget class="QLineEdit" name="hisDisplay">
      <property name="font">
       <font>
        <pointsize>12</pointsize>
        <fontweight>Light</fontweight>
       </font>
      </property>
      <property name="cursor">
       <cursorShape>ArrowCursor</cursorShape>
      </property>
      <property name="focusPolicy">
       <enum>Qt::FocusPolicy::StrongFocus</enum>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LayoutDirection::RightToLeft</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: transparent;
border: 2px solid transparent;
</string>
      </property>
      <property name="inputMask">
       <string notr="true"/>
      </property>
      <property name="text">
       <string notr="true"/>
      </property>
      <property name="frame">
       <bool>false</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLineEdit" name="txtDisplay">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>42</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>42</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>16</pointsize>
       </font>
      </property>
      <property name="cursor">
       <cursorShape>ArrowCursor</cursorShape>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LayoutDirection::RightToLeft</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: transparent;
border: 2px solid transparent;</string>
      </property>
      <property name="inputMask">
       <string notr="true"/>
      </property>
      <property name="text">
       <string notr="true"/>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
      </property>
      <property name="readOnly">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>40</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>18</pointsize>
        <stylestrategy>NoAntialias</stylestrategy>
        <kerning>false</kerning>
       </font>
      </property>
      <property name="text">
       <string>Calculator</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <layout class="QGridLayout" name="gridLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
      </property>
      <property name="leftMargin">
       <number>2</number>
      </property>
      <property name="topMargin">
       <number>2</number>
      </property>
      <property name="rightMargin">
       <number>2</number>
      </property>
      <property name="bottomMargin">
       <number>2</number>
      </property>
      <property name="spacing">
       <number>2</number>
      </property>
      <item row="2" column="0">
       <widget class="QPushButton" name="button_num_1">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">1</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">numButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="4" column="2">
       <widget class="QPushButton" name="button_num_6">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">6</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">numButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QPushButton" name="button_num_7">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">7</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">numButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QPushButton" name="button_num_8">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">8</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">numButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="button_num_2">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">2</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">numButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QPushButton" name="button_num_0">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">0</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">numButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QPushButton" name="button_backspace">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>7</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">Backspace</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="button_CE">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">CE</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="button_num_3">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>1</width>
          <height>1</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>70</width>
          <height>70</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">3</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">numButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="5" column="2">
       <widget class="QPushButton" name="button_num_9">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">9</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">numButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QPushButton" name="button_num_4">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">4</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">numButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QPushButton" name="button_num_5">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">5</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">numButtonGroup</string>
        </attribute>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="button_factorial">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">n!</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QPushButton" name="button_add">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>1</width>
          <height>1</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>70</width>
          <height>70</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">+</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QPushButton" name="button_subtract">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>1</width>
          <height>1</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>70</width>
          <height>70</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">−</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="4" column="3">
       <widget class="QPushButton" name="button_multiply">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>1</width>
          <height>1</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>70</width>
          <height>70</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">×</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="5" column="3">
       <widget class="QPushButton" name="button_divide">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>1</width>
          <height>1</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>70</width>
          <height>70</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>14</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">÷</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="6" column="3">
       <widget class="QPushButton" name="button_equal">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>1</width>
          <height>1</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>70</width>
          <height>70</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <bold>true</bold>
          <stylestrategy>PreferDefault</stylestrategy>
          <kerning>true</kerning>
         </font>
        </property>
        <property name="acceptDrops">
         <bool>false</bool>
        </property>
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(0, 85, 127); color: white</string>
        </property>
        <property name="text">
         <string notr="true">=</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="button_absolute">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">|x|</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="button_close_parenthesis">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">)</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="button_open_parenthesis">
        <property name="minimumSize">
         <size>
          <width>70</width>
          <height>42</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
          <stylestrategy>NoAntialias</stylestrategy>
          <kerning>false</kerning>
         </font>
        </property>
        <property name="text">
         <string notr="true">(</string>
        </property>
        <property name="iconSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>button_num_1</tabstop>
  <tabstop>button_num_2</tabstop>
  <tabstop>button_num_3</tabstop>
  <tabstop>button_num_4</tabstop>
  <tabstop>button_num_5</tabstop>
  <tabstop>button_num_6</tabstop>
  <tabstop>button_num_7</tabstop>
  <tabstop>button_num_8</tabstop>
  <tabstop>button_num_9</tabstop>
  <tabstop>button_num_0</tabstop>
 </tabstops>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="numButtonGroup"/>
 </buttongroups>
</ui>
